### PR TITLE
feat: Present Proof - functions ExecuteInbound/Outbound were merged

### LINKED ISSUE #1528: Present Proof - state interface needs to be changed
State interface needs to be changed to: 

```
type state interface {
	Name() string
	CanTransitionTo(next state) bool
	Execute(msg *metaData) (state, stateAction, error)
}
```

### PR DESCRIPTION
**Pull request by soluchok**
Functions ExecuteInbound/Outbound were merged.

Closes #1528 
Signed-off-by: Andrii Soluk <isoluchok@gmail.com>

### CODE REVIEW

File: pkg/didcomm/protocol/presentproof/states.go [hunk 1]
```diff
@@ -202,28 +181,25 @@ func (s *requestSent) CanTransitionTo(st state) bool {
...
 	}
+}
 
-	// creates the state's action
-	action := func(messenger service.Messenger) error {
-		// sets message type
-		md.request.Type = RequestPresentationMsgType
-		return messenger.ReplyTo(md.Msg.ID(), service.NewDIDCommMsgMap(md.request))
+func (s *requestSent) Execute(md *metaData) (state, stateAction, error) {
+	if !md.inbound {
```
Comment (llorllale) [inline pkg/didcomm/protocol/presentproof/states.go]: curious: I see we're still using directionality here. Did you consider using role names instead?

Comment (llorllale) [reply pkg/didcomm/protocol/presentproof/states.go]: related: is there a followup plan to also merge the two Handle* service functions?

Comment (soluchok) [reply pkg/didcomm/protocol/presentproof/states.go]: I did not think about it. `Handle*` functions have a common `service` interface. I can try to get rid of `inbound`.

Comment (soluchok) [reply pkg/didcomm/protocol/presentproof/states.go]: @llorllale I've merged Handle* functions and also removed `inbound` flag.

File: pkg/didcomm/protocol/presentproof/states.go [hunk 2]
```diff
@@ -297,28 +265,24 @@ func (s *proposalSent) CanTransitionTo(st state) bool {
 		st.Name() == stateNameAbandoning
 }
 
-func (s *proposalSent) ExecuteInbound(md *metaData) (state, stateAction, error) {
+func canReplyTo(msg service.DIDCommMsgMap) bool {
+	_, ok := msg["~thread"]
```
Comment (troyronda) [inline pkg/didcomm/protocol/presentproof/states.go]: "~thread" should be a constant.

**Pull request merged**